rmgpy/rmg/model.py

```diff
@@ -795,7 +798,35 @@ def processNewReactions(self, newReactions, newSpecies, pdepNetwork=None):
                         self.core.reactions.remove(rxn)
                     if rxn in self.edge.reactions:
                         self.edge.reactions.remove(rxn)
-
+            
+            if not numpy.isinf(self.toleranceThermoKeepSpeciesInEdge) and spcs != []: #do thermodynamic filtering
+            
```

Comment:
I am wondering if this could be a separate, possibly private (or not depending on your preference) method. Thoughts?